apps/web/modules/api/v2/management/surveys/[surveyId]/contact-links/segments/[segmentId]/lib/contact.ts

```diff
@@ -0,0 +1,138 @@
+import { getSegment } from "@/modules/api/v2/management/surveys/[surveyId]/contact-links/segments/[segmentId]/lib/segment";
+import { getSurvey } from "@/modules/api/v2/management/surveys/[surveyId]/contact-links/segments/[segmentId]/lib/surveys";
+import { sanitizeFields } from "@/modules/api/v2/management/surveys/[surveyId]/contact-links/segments/[segmentId]/lib/utils";
+import { TContactWithAttributes } from "@/modules/api/v2/management/surveys/[surveyId]/contact-links/segments/[segmentId]/types/contact";
+import { ApiErrorResponseV2 } from "@/modules/api/v2/types/api-error";
+import { ApiResponseWithMeta } from "@/modules/api/v2/types/api-success";
+import { segmentFilterToPrismaQuery } from "@/modules/ee/contacts/segments/lib/filter/prisma-query";
+import { prisma } from "@formbricks/database";
+import { logger } from "@formbricks/logger";
+import { Result, err, ok } from "@formbricks/types/error-handlers";
+
+export const getContactsInSegment = async (
+  surveyId: string,
+  segmentId: string,
+  limit: number,
+  skip: number,
+  attributeKeys?: string
+): Promise<Result<ApiResponseWithMeta<TContactWithAttributes[]>, ApiErrorResponseV2>> => {
+  try {
+    const surveyResult = await getSurvey(surveyId);
+    if (!surveyResult.ok) {
+      return err(surveyResult.error);
+    }
+
+    const survey = surveyResult.data;
+
+    if (survey.type !== "link") {
+      const error: ApiErrorResponseV2 = {
+        type: "forbidden",
+        details: [{ field: "surveyId", issue: "Invalid survey" }],
+      };
+      return err(error);
+    }
+
+    if (survey.status !== "inProgress") {
+      const error: ApiErrorResponseV2 = {
+        type: "forbidden",
+        details: [{ field: "surveyId", issue: "Invalid survey" }],
+      };
+      return err(error);
+    }
+
+    const segmentResult = await getSegment(segmentId);
+    if (!segmentResult.ok) {
+      return err(segmentResult.error);
+    }
+
+    const segment = segmentResult.data;
+
+    if (survey.environmentId !== segment.environmentId) {
+      logger.error({ surveyId, segmentId }, "Survey and segment are not in the same environment");
+      const error: ApiErrorResponseV2 = {
+        type: "bad_request",
+        details: [{ field: "segmentId", issue: "Environment mismatch" }],
+      };
+      return err(error);
+    }
+
+    const segmentFilterToPrismaQueryResult = await segmentFilterToPrismaQuery(
+      segment.id,
+      segment.filters,
+      segment.environmentId
+    );
+
+    if (!segmentFilterToPrismaQueryResult.ok) {
+      return err(segmentFilterToPrismaQueryResult.error);
+    }
+
+    const { whereClause } = segmentFilterToPrismaQueryResult.data;
+
+    const attributesToInclude = sanitizeFields(attributeKeys);
+    const allowedAttributes = attributesToInclude.slice(0, 20);
+
+    const [totalContacts, contacts] = await prisma.$transaction([
```

Comment:
just a question, maybe a nitpick also, but can we cache this?
if not exactly this transaction, maybe we can cache the outer function `getContactsInSegment`